packages/vue-components/src/Pic.vue

```diff
@@ -53,20 +59,30 @@ export default {
       }
       return this.widthFromHeight;
     },
+    computedHeight() {
+      return this.heightFromWidth;
+    },
+    computedLoadType() {
+      return this.lazy ? 'lazy' : 'eager';
+    },
   },
   data() {
     return {
       widthFromHeight: '',
+      heightFromWidth: '',
     };
   },
   methods: {
-    computeWidth() {
-      if (!this.hasWidth && this.hasHeight) {
-        const renderedImg = this.$refs.pic;
-        const imgHeight = renderedImg.naturalHeight;
-        const imgWidth = renderedImg.naturalWidth;
-        const aspectRatio = imgWidth / imgHeight;
+    computeWidthAndHeight() {
+      const renderedImg = this.$refs.pic;
+      const imgHeight = renderedImg.naturalHeight;
+      const imgWidth = renderedImg.naturalWidth;
+      const aspectRatio = imgWidth / imgHeight;
+      if (this.hasWidth) { // if width is present, overwrite the height (if any) to maintain aspect ratio
```

Comment:
Hmmm if this behavior necessary? I think its ok for a user to adjust the aspect ratio as needed. 
This might also be breaking behavior in that case.